Changes arduino-common to arduino-atmega-common.

antmerlino opened the pull request:
Moves atmega common code from board files into atmega-common. 
Merges atmega2560 and atmega-common platforms to simplify structure

mali commented on boards/arduino-atmega-common/include/arduino_pinmap.h:
```diff
@@ -32,6 +34,40 @@ extern "C" {
  */
+
+/* Digital pins */
+#ifdef CPU_ATMEGA328P
```
Where does this define come from ?
Maybe better use avr-lib \_\_AVR_ATmega328P\_\_ defines ?

antmerlino replied:
This should come from the RIOT defines for CPU

kYc0o replied:
I mean, what this PR propose.

mali commented on boards/arduino-atmega-common/include/periph_conf.h:
```diff
@@ -1,21 +1,22 @@
  *
  * @author      Hinnerk van Bruinehsen <h.v.bruinehsen@fu-berlin.de>
- * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Laurent Navet <laurent.navet@gmail.com>
```
re-add Hauke Petersen <hauke.petersen@fu-berlin.de> here

kYc0o commented:
Please rebase and address the comment of @mali in periph_conf.h. After I think this is ready.

Thanks @antmerlino for the factorisation!

antmerlino commented:
Done!

The only thing I see that may be a concern is it seems like the permissions of some of the renamed dist/ files changed for arduino-atmega-common... Is this a problem? If so I'm not 100% sure how to correct it in the commit. Any advice?

kYc0o commented:
I don't think that the permissions would be an issue.

However, looking at the commit, it seems very big, at a first sight it looked ok but we have a policy of committing what we change by prefixing the directory where the commit applies, for instance "boards/arduino-atmega-common: move periph config" and so on, depending on what are you doing on the file... I know it's a bit picky but it's for the sake of organisation.

Can you prefix and split the commits following this notation?

Thanks again!

antmerlino commented:
@kYc0o Splitting the commits is going to be a little challenging because of the name change and merging of the platform.  I can probably do it but it's going to take me a while.  Is there anyway we can let this one slide and I'll be sure to split next time?

If not I suppose I can do it, like I said, it's just going to be a huge PITA.

Let me know! Thanks

kYc0o commented on boards/arduino-atmega-common/Makefile.include:
```diff
@@ -0,0 +1,29 @@
+endif
+
+export TERMPROG = $(RIOTBASE)/dist/tools/pyterm/pyterm
+export TERMFLAGS = -b 9600 -p $(PORT)
```
I think this `TERMFLAGS` should belong to each board.

antmerlino replied:
Should we just change it to ?= then.  That way the default still works without any of the current arduino boards specifying otherwise.  The reason I moved it is because currently all the boards that are considered arduino-atmega boards were using the same settings.

kYc0o replied:
OK I understand, the `?=` seems a good option 👍

kYc0o commented on boards/arduino-atmega-common/Makefile.include:
```diff
@@ -0,0 +1,29 @@
+ifeq ($(OS),Linux)
+		PORT = $(LINUX_PORT)
+else ifeq ($(OS),Darwin)
+  PORT ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
```
Please define a `$(DARWIN_PORT)` here too for each board.

kYc0o commented on boards/waspmote-pro/Makefile.include:
```diff
@@ -43,14 +43,5 @@ ifeq ($(PROGRAMMER), stk500v1)
-
-export CFLAGS += $(CFLAGS_CPU) $(CFLAGS_LINK) $(CFLAGS_DBG) $(CFLAGS_OPT)
-export ASFLAGS += $(CFLAGS_CPU) $(CFLAGS_DBG)
-export LINKFLAGS += $(CFLAGS_CPU) $(CFLAGS_DBG) $(CFLAGS_OPT) -e reset_handler
```
I'd like to have these definitions on the board's Makefile.include, even though this board is using an atmega MCU, it's not an arduino.

antmerlino replied:
These flags got moved to the atmega_common not to the arduino-atmega-common.  At the end of the day flags like this are really cpu specific not board specific.  

Let me know if you agree.

kYc0o replied:
OK, got it. Then it should belong to a separated commit, since the naming I proposed won't work with this board, especially because it's being moved to cpu/atmega_common.

antmerlino commented:
@kYc0o Alright, I made a couple changes to address your concerns.  I split the commit into smaller chunks.  I also noticed a few of the arduino platforms were using LINUX_PORT, but everywhere else uses PORT_LINUX, so I adjusted all arduino boardsd to comply with the rest of RIOT.  I moved the defaults for the PORT_LINUX and PORT_DARWIN to the board specific files.  Though, we may consider moving those lines for the platforms dependent on arduino-atmega-common and just make the defaults in the common framework.  Either way, hopefully this is good now.

kYc0o commented:
@antmerlino Changes look good so far, I'll test this asap and then try to merge.

Sorry for being so picky, but we want some consistency in the structure of RIOT, even if we have much work to do on that matter.

I'll trigger the CI to see if there are not broken things and after my tests I'll give my ACK for this PR.

Thanks a lot for your efforts and patience!

mali commented:
tested with uno board :+1:

kYc0o commented:
Everything is OK, tested with mega2560 and waspmote. GO!

kYc0o commented:
@antmerlino Just for the record, for next PRs you can take a quick look to our [coding conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions) and [development procedures](https://github.com/RIOT-OS/RIOT/wiki/Development-procedures), which will help you to create nice PRs.

Thanks again for your great work!
